server/src/assets/game/tableGenerator/converter.py

```diff
@@ -1,127 +1,193 @@
-            signal = signal + "a"
-            if signal in destinations:
-                destinationsSorted.append(signal)
-
-    resultData[block] = {}
-    for destination in destinationsSorted:
-        resultData[block][destination] = {}
-        resultData[block][destination]["route-id"] = originalResultData[block][destination]["route-id"]
-        resultData[block][destination]["orientation"] = originalResultData[block][destination]["orientation"]
-        resultData[block][destination]["block"] = originalResultData[block][destination]["block"]
-        resultData[block][destination]["segment"] = originalResultData[block][destination]["segment"]
-
-with open("../destinations-swtbahn-full.json", "w") as file:
-    file.write("const allPossibleDestinationsSwtbahnFull = ")
-    file.write(json.dumps(resultData, indent=2))
-    file.write(";")
+        else:
+            print("Error with BlockID {}".format(block["id"]))
+            isError = True
+        if isError:
+            print("error with {}".format(route))
+            break
+        resultData[block["id"]][destination] = {}
+        resultData[block["id"]][destination]["route-id"] = routeID
+        resultData[block["id"]][destination]["orientation"] = orientation
+        resultData[block["id"]][destination]["block"] = lastBlock
+        resultData[block["id"]][destination]["segment"] = stopSegment
+
+
+    
+for entry in configFolderItemList: # Get a list of configuration possibilities which are given by the model railway
+    if entry.is_dir():
+        folderList.append(entry.name)
+
+for configuration in folderList: # Roll over the list of configuration possibilities and check if a mapping CSV is existing
+    
+    mappingFolderContent = os.scandir(groupingFileDirectory)
+    flagMappings = []
+    for entry in mappingFolderContent:
+        if entry.is_file() and entry.name[-4:] == ".csv":
+            flagMappings.append(entry.name[:-4])
+
+    if configuration not in flagMappings: # Ignore all entries which doesnt have an equivalent Mapping-CSV
+        continue
+    
+    
+    blacklist = []
+
+    if os.path.exists("{}/{}.txt".format(blacklistFileDirectory, configuration)): # import the blacklists if exists
+        for line in open("{}/{}.txt".format(blacklistFileDirectory, configuration)):
+            blacklist.append(int(line.strip()))        
+
+
+    interlockingTableFile = "{}/{}/{}".format(pathToConfig, configuration, interlockingTableFileName)
+    configurationBahnFile = "{}/{}/{}".format(pathToConfig, configuration, configurationBahnFileName)
+
+
+    interlockingTable = json.loads(json.dumps(yaml.safe_load(open(interlockingTableFile))))
+    configuratonBahn = json.loads(json.dumps(yaml.safe_load(open(configurationBahnFile))))
+    
+    groupingFile = "{}/{}.csv".format(groupingFileDirectory, configuration)
+
+
+    resultData = {}
+
+    blocktypes = ["blocks", "platforms"]
+
+    for blockType in blocktypes: # Roll over all Blocktypes
```

Comment:
What is the goal of this loop? Could be simplified if I knew its goal...

Reply:
loop over the two blocktypes which we currrently have in our interlocking file (platforms + blocks), but also has the option to extend it, if we need.